### PR TITLE
docs(docker): add ghworkflows example for GAR with Workload Identity

### PR DESCRIPTION
**Pull request by jazzlyn**
## Changes

update documentation to show a proper example of docker data source configuration with GitHub workflows, workload identity and Google artifact registry.

## Context

https://github.com/renovatebot/renovate/discussions/29618#discussioncomment-9761641

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [x] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

### CODE REVIEW

**Review by rarkins — DISMISSED** [review]

File: docs/usage/docker.md [hunk 1]
```diff
@@ -279,12 +279,44 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
 
-Just configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) /
-[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) as normal and _don't_
-provide a username, password or token. Renovate will automatically retrieve the credentials using the
-google-auth-library.
+[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) has to be configured and the Service Account needs `artifactregistry.repositories.downloadArtifacts` permission.
+
+###### With Application Default Credentials (Self-Hosted only)
+
+Just configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal and _don't_ provide a username, password or token.
```
Comment (HonkingGoose) [inline docs/usage/docker.md:288]:
```suggestion
Configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal.
Do not provide a username, password or token.
```

File: docs/usage/docker.md [hunk 2]
```diff
@@ -279,12 +279,44 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
+Renovate will automatically retrieve the credentials using the google-auth-library.
```
Comment (HonkingGoose) [inline docs/usage/docker.md:289]:
```suggestion
Renovate will get the right credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).
```

Please check if I'm linking to the correct package!

File: docs/usage/docker.md [hunk 3]
```diff
@@ -279,12 +279,44 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
 #### Google Container Registry / Google Artifact Registry
 
-##### Using Application Default Credentials / Workload Identity (Self-Hosted only)
+##### Using Workload Identity
 
-Just configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) /
-[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) as normal and _don't_
-provide a username, password or token. Renovate will automatically retrieve the credentials using the
-google-auth-library.
+[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) has to be configured and the Service Account needs `artifactregistry.repositories.downloadArtifacts` permission.
```
Comment (HonkingGoose) [inline docs/usage/docker.md:284]:
```suggestion
[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) must be configured and the Service Account must have the `artifactregistry.repositories.downloadArtifacts` permission.
```

File: docs/usage/docker.md [hunk 4]
```diff
@@ -279,12 +279,44 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
+
+###### With Application Default Credentials (Self-Hosted only)
+
+Just configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal and _don't_ provide a username, password or token.
+Renovate will automatically retrieve the credentials using the google-auth-library.
+
+###### With short-lived access token / GitHub Actions (Self-hosted only)
+
+Below you find the example configuration for both Workload Identity and the Renovate Host rules. For a full GitHub Workflow example see [renovatebot/github-action](https://github.com/renovatebot/github-action) repo.
```
Comment (HonkingGoose) [inline docs/usage/docker.md:293]:
```suggestion
Example configuration for the Workload Identity and Renovate's host rules:
```

File: docs/usage/docker.md [hunk 5]
```diff
@@ -279,12 +279,44 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
+          matchHost: "us-central1-docker.pkg.dev",
+          hostType: "docker",
+          username: "oauth2accesstoken",
+          password: "${{ steps.auth.outputs.access_token }}"
+        }
+      ]
+  with:
+    token: ${{ secrets.RENOVATE_TOKEN }}
+    configurationFile: .github/renovate.json5
+```
```
Comment (HonkingGoose) [inline docs/usage/docker.md:320]:
```suggestion

You can find a [full GitHub Workflow example on the `renovatebot/github-action` repository](https://github.com/renovatebot/github-action).
```

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
First round of review.

File: docs/usage/docker.md [hunk 6]
```diff
@@ -279,12 +279,47 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
-[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) as normal and _don't_
-provide a username, password or token. Renovate will automatically retrieve the credentials using the
-google-auth-library.
+[Workload Identity](https://cloud.google.com/kubernetes-engine/docs/how-to/workload-identity) must be configured and the Service Account must have the `artifactregistry.repositories.downloadArtifacts` permission.
+
+###### With Application Default Credentials (Self-Hosted only)
+
+Configure [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc) as normal.
+Do not provide a username, password or token.
+Renovate will get the credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).
```
Comment (HonkingGoose) [inline docs/usage/docker.md:290]:
```suggestion
To let Renovate authenticate with [ADC](https://cloud.google.com/docs/authentication/provide-credentials-adc), you must:

- Configure ADC as normal
- _Not_ provide a username, password or token

Renovate will get the credentials with the [`google-auth-library`](https://www.npmjs.com/package/google-auth-library).
```

File: docs/usage/docker.md [hunk 7]
```diff
@@ -279,12 +279,47 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
+
+###### With short-lived access token / GitHub Actions (Self-hosted only)
```
Comment (HonkingGoose) [inline docs/usage/docker.md:292]:
```suggestion
###### With short-lived access token / GitHub Actions (self-hosted only)
```

File: docs/usage/docker.md [hunk 8]
```diff
@@ -279,12 +279,47 @@ To make use of this authentication mechanism, specify the username as `AWS`:
...
+
+Example configuration for Workload Identity and Renovate's host rules:
+
+```yaml
```
Comment (HonkingGoose) [inline docs/usage/docker.md:296]: Please:

1. Delete line 294: `Example configuration for Workload Identity and Renovate's host rules:`.
2. Add a title to the YAML codeblock. [^1]

The bit after the three backticks should look like this:

`yaml title="Example for Workload Identity plus Renovate host rules"`

[^1]: [Material for MkDocs, adding a title to a codeblock](https://squidfunk.github.io/mkdocs-material/reference/code-blocks/#adding-a-title)

File: docs/usage/docker.md [hunk 9]
```diff
@@ -386,7 +421,7 @@ If you have dependencies on Google Container Registry (and Artifact Registry) yo
       }
       ```
 
-##### Using short-lived access tokens
+##### Using short-lived access token / Gitlab CI / Google Cloud
```
Comment (HonkingGoose) [inline docs/usage/docker.md:424]: This section feels like a wall of text. Can you add some headings to break it up into smaller parts?

**Review by HonkingGoose — CHANGES_REQUESTED** [review]
I think a list of steps is easier to read and skim than the plain sentences we have now. Also we should break up the wall of text at the bottom.

File: docs/usage/docker.md (hunk 9 above)
Comment (jazzlyn) [reply docs/usage/docker.md:428]: out of scope for this pr

Comment (HonkingGoose) [reply docs/usage/docker.md:428]: I'll make a follow up PR later then.

**Review by HonkingGoose — APPROVED** [review]
I'm happy with the docs. We still need a maintainer to check the technical accuracy of the text though.

**Review by rarkins — APPROVED** [review]

**Pull request merged**